lnrpc/routerrpc/router.proto

```diff
@@ -59,24 +59,54 @@ message PaymentRequest {
     int32 cltv_limit = 9;
 }
 
-message PaymentResponse {
+message SendPaymentResponse {
+}
+
+message LookupPaymentRequest {
+    /// The hash of the payment to look up.
+    bytes payment_hash = 1;
+}
+
+enum PaymentState {
     /**
-    The payment hash that we paid to. Provided so callers are able to map
-    responses (which may be streaming) back to their original requests.
+    Payment is still in flight.
     */
-    bytes pay_hash = 1;
+    IN_FLIGHT = 0;
 
     /**
-    The pre-image of the payment successfully completed.
+    Payment completed successfully.
     */
-    bytes pre_image = 2;
+    SUCCEEDED = 1;
 
     /**
-    If not an empty string, then a string representation of the payment error.
+    There are more routes to try, but the payment timeout was exceeded.
     */
-    string payment_err = 3;
+    FAILED_TIMEOUT = 2;
```

Comment:
We could also do `FAILED` with extra `Reason` field?

Reply:
Like this it isn't possible to mistakingly return `FAILED` without a reason.